Sources/protoc-gen-swift/MessageGenerator.swift

```diff
@@ -279,8 +279,11 @@ class MessageGenerator {
       generateWithLifetimeExtension(printer: &p, throws: true) { p in
         p.print("while let \(varName) = try decoder.nextFieldNumber() {\n")
         p.indent()
-        if !fields.isEmpty {
-
+        // If a message only has extensions and there are multiple of extension
+        // ranges; get more compact codegen by still using the `switch..case`
+        // code. This also avoids typechecking perforamnce issues if there are
+        // dozens of ranges due to to the comparisions to the literals.
```

Comment:
```suggestion
        // If a message only has extensions and there are multiple extension
        // ranges, get more compact source gen by still using the `switch..case`
        // code. This also avoids typechecking performance issues if there are
        // dozens of ranges because we aren't constructing a single large
        // expression containing untyped integer literals.
```

Reply:
done